src/java_bytecode/java_bytecode_convert_method.cpp

```diff
@@ -1416,7 +1417,11 @@ codet java_bytecode_convert_methodt::convert_instructions(
     else if(statement==patternt("?ipush"))
     {
       assert(results.size()==1);
-      results[0]=typecast_exprt(arg0, java_int_type());
+      namespacet ns(symbol_table);
+      // Force a simplify here in case we have --no-simplify
+      // because the simplifier is the only pass that knows how to
+      // deal with this cast from integer_typet:
+      results[0]=simplify_expr(typecast_exprt(arg0, java_int_type()), ns);
```

Comment:
Pushed an amended version that creates an integer directly in bytecode-convert-method. I didn't do it in the parser because that uses integer_typet everywhere, and making particular exceptions seemed uglier. Rather let's say the parser turns the bytecode format into a tree where integer members are represented as mp_integers (effectively), then convert-method produces valid codet expressions (and typecast-integer-to-bitvector is specifically not one).